tests/providers/microsoft/azure/hooks/test_azure_fileshare.py

```diff
@@ -112,6 +113,20 @@ def test_list_directories_and_files(self, mock_service):
         hook.list_directories_and_files('share', 'directory', timeout=1)
         mock_instance.list_directories_and_files.assert_called_once_with('share', 'directory', timeout=1)
 
+    @mock.patch('airflow.providers.microsoft.azure.hooks.azure_fileshare.FileService', autospec=True)
+    def test_list_files(self, mock_service):
+        mock_instance = mock_service.return_value
+        mock_instance.list_directories_and_files.return_value = [
+            File("file1"),
+            File("file2"),
+            Directory("dir1"),
+            Directory("dir2"),
+        ]
+        hook = AzureFileShareHook(wasb_conn_id='wasb_test_sas_token')
+        files = hook.list_files('share', 'directory', timeout=1)
```

Comment:
Should it be a mock also?

Reply:
its not - `list_files` is a wrapper around original `list_files_and_directories` api call which makes the original call and then filters out directories from results list, therefore mocking `FileService.list_files_and_directories` is enough